hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/TimestampBasedAvroKeyGenerator.java

```diff
@@ -125,7 +126,7 @@ public TimestampBasedAvroKeyGenerator(TypedProperties config) throws IOException
 
   @Override
   public String getPartitionPath(GenericRecord record) {
-    Object partitionVal = HoodieAvroUtils.getNestedFieldVal(record, getPartitionPathFields().get(0), true);
+    Object partitionVal = HoodieAvroUtils.getNestedFieldVal(record, getPartitionPathFields().get(0), true, isConsistentLogicalTimestampEnabled());
```

Comment:
can't we fetch the value of isConsistentLogicalTimestampEnabled in the constructor from props?
```
TimestampBasedKeyGenerator(TypedProperties config)
```
We should be very cautious in changing public apis. KeyGenerator apis are public and likely users have their custom key gen.

Reply:
Not changing the keygen API here. I am using the config and class hierarchy itself. `isConsistentLogicalTimestampEnabled()` is defined in `BaseKeyGenerator` superclass for reusability.